lib/guard/yard/server.rb

```diff
@@ -58,6 +50,30 @@ def verify
         )
         false
       end
+
+      private
+
+      def windows_kill
+        # /t includes child process
+        # /f forces (required else we'll fail)
+        if system("taskkill /f /t /pid #{pid}").nil?
+          raise "Cannot kill server: Windows' taskkill command not found"
+        end
+
+        # Process was killed or it didn't exist (something else..?).
+        true
+      end
+
+      def posix_kill
+        begin
+          Process.kill('QUIT', pid)
+          Process.wait2(pid)
+        # rubocop:disable Lint/HandleExceptions
+        rescue Errno::ESRCH, Errno::ECHILD
+          # Process is already dead.
+        end
```

Comment:
I think the `begin` clause is no longer necessary - it an be added to the method level now.

Reply:
If that's the case then you need to return `true` twice (like before) since if the exception is caught the method will return `nil`. But, I guess your saying this is okay.

Reply:
I was envisioning this because we never return `false`, just raise exceptions:

```
def kill
  ...
  Gem.win_platform? ? windows_kill : posix_kill
  true
end

def windows_kill
  ...
  # no explicit return
end

def posix_kill
  Process.kill('QUIT', pid)
  Process.wait2(pid)
rescue Errno::ESRCH, Errno::ECHILD
  true # not strictly necessary either
end
```

Not particularly important though and easy enough to refactor later if necessary. The current implementation does make it easier to return `false` in the future for some reason.